services/explorer/graphql/server/graph/fetcher.go

```diff
@@ -0,0 +1,498 @@
+			return nil, fmt.Errorf("no cctp log found with request id %s", kappa)
+		}
+		bridgeEvent, err = r.Parsers.CCTParsers[chainID].MatureLogs(ctx, ifaceCctpEvent.CCTPEvent, ifaceCctpEvent.IFace, chainID)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("could not mature logs: %w", err)
+	}
+	return bridgeEvent, nil
+}
+
+// parseSwapLog this is a swap event, we need to get the address from it.
+func (r Resolver) parseSwapLog(ctx context.Context, swapLog ethTypes.Log, chainID uint32) (*swapReplacementData, error) {
+	// parse swap with swap filter
+	var swapReplacement swapReplacementData
+	filterKey := fmt.Sprintf("%d_%s", chainID, swapLog.Address.String())
+	filter := r.SwapFilters[filterKey]
+	if filter == nil {
+		return nil, fmt.Errorf("this swap address is not in the server config, chainid: %d, server: %s", chainID, swapLog.Address.String())
+	}
+	swapEvent, err := filter.ParseTokenSwap(swapLog)
+	if err != nil || swapEvent == nil {
+		return nil, fmt.Errorf("error parsing log, chainid: %d, server: %s", chainID, swapLog.Address.String())
+	}
+
+	iFace, err := filter.ParseTokenSwap(swapLog)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse swap event: %w", err)
+	}
+	soldID := iFace.SoldId
+	address, err := r.DB.GetString(ctx, fmt.Sprintf("SELECT token_address FROM token_indices WHERE contract_address='%s' AND chain_id=%d AND token_index=%d", swapLog.Address.String(), chainID, soldID.Uint64()))
+	if err != nil {
+		return nil, fmt.Errorf("could not parse swap event: %w", err)
+	}
+	swapReplacement = swapReplacementData{
+		Amount:  iFace.TokensSold,
+		Address: common.HexToAddress(address),
+	}
+	return &swapReplacement, nil
+}
+
+func (r Resolver) checkKappaExists(ctx context.Context, kappa string, chainID uint32) bool {
+	var kappaBytes32 [32]byte
+
+	kappaBytes := common.Hex2Bytes(kappa)
+	copy(kappaBytes32[:], kappaBytes)
+
+	exists, err := r.Refs.BridgeRefs[chainID].KappaExists(&bind.CallOpts{
+		Context: ctx,
+	}, kappaBytes32)
+	if err != nil {
+		logger.Errorf("Could not check if kappa exists on chain %d. Error: %v", chainID, err)
+		return false
+	}
+	return exists
+}
+
+func (r Resolver) checkRequestIDExists(ctx context.Context, requestID string, chainID uint32) bool {
+	var kappaBytes32 [32]byte
+	kappaBytes := common.Hex2Bytes(requestID)
+	copy(kappaBytes32[:], kappaBytes)
+	exists, err := r.Refs.CCTPRefs[chainID].IsRequestFulfilled(&bind.CallOpts{
+		Context: ctx,
+	}, kappaBytes32)
+	if err != nil {
+		logger.Errorf("Could not check if request exists on chain %d. Error: %v", chainID, err)
+		return false
+	}
+	return exists
+}
+
+func (r Resolver) storeBridgeEvent(bridgeEvent interface{}) {
+	storeCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	storeErr := r.DB.StoreEvent(storeCtx, bridgeEvent)
+	if storeErr != nil {
+		logger.Errorf("could not store log while storing origin bridge watcher tx %v", storeErr)
+	}
+}
```

Comment:
this is a bit more complicated

this function is used in a separate go routine to store newly indexed events into the database. 
the purpose of the upstream function is to return data from the chain and return it to the user (if scribe/explorer have not picked it up yet), if storage fails it should not disrupt this function's core purpose. Furthermore, we can assume that in the case that storage of this data fails, it will be picked up by scribe and explorer in the next minute and stored correctly. 

enforcing storage w/retries would only speed up user queries for getting destination queries in the <≈40 seconds between the tx landing on the destination chain and scribe + explorer getting it. Because each one of the routines could end up in long running go routines, I think it is reasonable to simply log the error and move on as this is a secondary and optional optimization for the core purpose of this function.